dspace-server-webapp/src/test/java/org/dspace/app/rest/ResearcherProfileRestRepositoryIT.java

```diff
@@ -0,0 +1,1935 @@
+
+        profile = context.reloadEntity(profile);
+
+        assertThat(getMetadataValues(profile, "person.identifier.orcid"), not(empty()));
+        assertThat(getMetadataValues(profile, "dspace.orcid.access-token"), not(empty()));
+        assertThat(getMetadataValues(profile, "dspace.orcid.refresh-token"), not(empty()));
+        assertThat(getMetadataValues(profile, "dspace.orcid.scope"), not(empty()));
+        assertThat(getMetadataValues(profile, "dspace.orcid.authenticated"), not(empty()));
+    }
+
+    @Test
+    public void testCloneFromExternalProfileAlreadyAssociated() throws Exception {
+
+        String id = user.getID().toString();
+        String authToken = getAuthToken(user.getEmail(), password);
+
+        getClient(authToken).perform(post("/api/eperson/profiles/").contentType(MediaType.APPLICATION_JSON_VALUE))
+                            .andExpect(status().isCreated()).andExpect(jsonPath("$.id", is(id)))
+                            .andExpect(jsonPath("$.visible", is(false))).andExpect(jsonPath("$.type", is("profile")));
+
+        getClient(authToken)
+            .perform(post("/api/eperson/profiles/").contentType(TEXT_URI_LIST)
+                                                   .content("http://localhost:8080/server/api/core/items/" + id))
+            .andExpect(status().isUnprocessableEntity());
+    }
+
+    private Item createProfile(EPerson ePerson) throws Exception {
+
+        String authToken = getAuthToken(ePerson.getEmail(), password);
+
+        AtomicReference<UUID> ePersonIdRef = new AtomicReference<UUID>();
+        AtomicReference<UUID> itemIdRef = new AtomicReference<UUID>();
+
+        getClient(authToken).perform(post("/api/eperson/profiles/")
+                                         .contentType(MediaType.APPLICATION_JSON_VALUE))
+                            .andExpect(status().isCreated())
+                            .andDo(result -> ePersonIdRef.set(fromString(read(result.getResponse().getContentAsString(),
+                                                                              "$.id"))));
+
+        getClient(authToken).perform(get("/api/eperson/profiles/{id}/item", ePersonIdRef.get())
+                                         .contentType(MediaType.APPLICATION_JSON_VALUE))
+                            .andExpect(status().isOk())
+                            .andDo(result -> itemIdRef.set(fromString(read(result.getResponse().getContentAsString(),
+                                                                           "$.id"))));
+
+        return itemService.find(context, itemIdRef.get());
+    }
+
+    private String getItemIdByProfileId(String token, String id) throws SQLException, Exception {
+        MvcResult result = getClient(token).perform(get("/api/eperson/profiles/{id}/item", id))
+                                           .andExpect(status().isOk())
+                                           .andReturn();
+
+        return readAttributeFromResponse(result, "$.id");
```

Comment:
Could we update the logic in this method to use the `.andDo(result -> idRef.set())` sort of approach to parse out the `id` similar to what you use just above this (on lines 1913-1916)?  From my experience, I've found that parsing the **entire** MvcResult is very slow & when done frequently it can slow down tests significantly.